### ISSUE TITLE
io::Result vs crossterm::Result

### ISSUE THREAD

**Issue by zrzka** [TimonPost/crossterm#229]
Look at the following example (`examples/terminal.rs`):

```rust
pub fn set_terminal_size() -> io::Result<()> {
    let terminal = terminal();
    terminal.set_size(10, 10)
}
```

I consider it a bit problematic for the crate user. `terminal.set_size(10, 10)` returns `crossterm::Result`:

```rust
/// The `crossterm` result type.
pub type Result<T> = std::result::Result<T, ErrorKind>;

/// Wrapper for all errors that can occur in `crossterm`.
#[derive(Debug)]
pub enum ErrorKind {
    IoError(io::Error),
    FmtError(fmt::Error),
    ResizingTerminalFailure(String),

    #[doc(hidden)]
    __Nonexhaustive,
}
```

As a user, I expect that it can return `ErrorKind::ResizingTerminalFailure`, `ErrorKind::FmtError`, etc. But the `set_terminal_size()` fn returns `io::Result`.

First question is why do we return `io::Result` here? It can be a bit unclear to the crate user.

Another problematic area is the `crossterm::Result` -> `io::Result` conversion:

```rust
impl From<ErrorKind> for io::Error {
    fn from(e: ErrorKind) -> io::Error {
        match e {
            ErrorKind::IoError(io) => return io,
            _ => io::Error::new(io::ErrorKind::Other, "can not convert error to IO error"),
        }
    }
}
```

In other words, if it's not an `ErrorKind::IoError`, the real error is discarded and replaced with a bit unhelpful error message: _can not convert error to IO error_.

## Questions

* Why do we use `io::Result` in examples when we have `crossterm::Result` & `crossterm::ErrorKind`?
* Shouldn't we just use `crossterm::Result` in all examples?
* What is the reason for `crossterm::Result` -> `io::Result` conversion?

I don't know the reasoning behind this, but I think that:

* All examples should use `crossterm::Result`
* `crossterm::Result` -> `io::Result` conversion should be removed

It seems a bit inconsistent to me. But as I said, I don't know the reasoning and I may be completely wrong.

**Comment by TimonPost** [comment]
So error management is not completed yet, when crossterm was developed I was very new to rust, so I used, of course, unwrap everywhere, as I learned more, I later introduced an error type. I removed a lot of unwraps and panics and made some changes to the API to accept `crossterm::Result`. 

The only reason that conversion is there is because of [here](https://github.com/TimonPost/crossterm/blob/master/crossterm_screen/src/screen/alternate.rs#L41) and here [here](https://github.com/TimonPost/crossterm/blob/master/crossterm_screen/src/screen/alternate.rs#L54)

So yes, I would prefer to have everything to be changed to `crossterm::Result` instead of `io::Result`. Then we can remove that conversion from `ErrorKind` to `io::Result`.  Also, I am not sure why I introduced that in the examples that are probably done a few months - year ago, but that can and should be changed to `crossterm::Result`. 

My original plan was to make a lot more errors to similar to `ResizingTerminalFailure`, but haven't got that far. Why I haven't finished with those, has to do with the reason that all ANSI codes are returning `IoError` and all WinAPI calls as well. It is a lot of work, to map all IO errors to our own ErrorKind.

**Comment by zrzka** [comment]
Thanks for the explanation. I have a free day tomorrow, so I'll spend it on this issue.

**Comment by Mooling0602** [comment]
🤔idk if I misunderstand this issue.

<img width="1080" height="2460" alt="Image" src="https://github.com/user-attachments/assets/103778d7-58d6-47b1-903c-5435b2a82362" />

<img width="1080" height="1904" alt="Image" src="https://github.com/user-attachments/assets/c39eead0-ddaf-48db-8f24-1ee33e230e11" />
